File: core/src/main/java/com/ustadmobile/core/view/UserProfileView.java
package com.ustadmobile.core.view;

import com.ustadmobile.core.db.UmProvider;
import com.ustadmobile.lib.db.entities.Person;


/**
 * Core View. Screen is for UserProfile's View
 */
public interface UserProfileView extends UstadView {


    // This defines the view name that is an argument value in the go() in impl.
    String VIEW_NAME = "UserProfile";

    //Any argument keys:

    /**
     * Method to finish the screen / view.
     */
    void finish();

    void updateToolbarTitle(String personName);

    void setLanguageSet(String languageSet);

    void updateImageOnView(String imagePath);

    void addImageFromCamera();

    void addImageFromGallery();


}

File: core/src/commonMain/kotlin/com/ustadmobile/core/view/UserProfileView.kt
package com.ustadmobile.core.view

import com.ustadmobile.lib.db.entities.Person

interface UserProfileView : LanguageOptionView  {

    fun finish()

    var person: Person

    fun updateToolbarTitle(personName: String)

    fun updateImageOnView(imagePath: String, skipCached: Boolean)

    fun callFinishAffinity()

    fun updateLastSyncedText(lastSynced: String)

    companion object {

        const val VIEW_NAME = "UserProfile"

        const val PERSON_UID = "person_uid"
    }
}
